host_core/lib/host_core/web_assembly/imports.ex

```diff
@@ -299,15 +300,83 @@ defmodule HostCore.WebAssembly.Imports do
       )
       |> perform_rpc_invoke(target_subject)
 
-    case unpack_invocation_response(invocation_res) do
-      {1, :host_response, msg} ->
-        Agent.update(agent, fn state -> %State{state | host_response: msg} end)
-        1
+    unpacked = unpack_invocation_response(invocation_res)
```

Comment:
nit: we don't use `unpacked` anywhere but the `case` so this function call can probably be used in the `case` statement instead